src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedInstanceConstructor.cs

```diff
@@ -266,5 +266,31 @@ internal sealed override DiagnosticInfo GetUseSiteDiagnostic()
             return ReturnType.TypeSymbol.GetUseSiteDiagnostic();
         }
         #endregion
+
+        protected void GenerateMethodBodyCore(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var factory = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            factory.CurrentFunction = this;
+            if (ContainingType.BaseTypeNoUseSiteDiagnostics is MissingMetadataTypeSymbol)
```

Comment:
>if (ContainingType.BaseTypeNoUseSiteDiagnostics is MissingMetadataTypeSymbol) [](start = 12, length = 77)

Can be moved to top of method so `factory.CloseMethod()` is not needed.

Reply:
I intentionally want to call `CloseMethod`, to put a method body (albeit an empty one) in such cases.
Method bodies get compiled before we decide whether to emit the injected types, and the method compiler doesn't like null bodies.

---
In reply to: [214983874](https://github.com/dotnet/roslyn/pull/29180#discussion_r214983874) [](ancestors = 214983874)